game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/sub/OffensiveSubsRetreatTest.java

```diff
@@ -0,0 +1,196 @@
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .over(true)
+                .build()),
+        Arguments.of(
+            "Defending Destroyer exists",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .defendingUnits(List.of(givenUnitDestroyer()))
+                .build()),
+        Arguments.of(
+            "Waiting to Die Defending Destroyer exists",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .defendingWaitingToDie(List.of(givenUnitDestroyer()))
+                .build()),
+        Arguments.of(
+            "Amphibious assault",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .gameData(
+                    MockGameData.givenGameData().withTransportCasualtiesRestricted(false).build())
+                .amphibious(true)
+                .build()),
+        Arguments.of(
+            "Can withdraw but only defenseless transports on the defense",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .gameData(
+                    MockGameData.givenGameData()
+                        .withTransportCasualtiesRestricted(true)
+                        .withSubmersibleSubs(false)
+                        .build())
+                .defendingUnits(List.of(givenUnitTransport()))
+                .attackerRetreatTerritories(List.of(mock(Territory.class)))
+                .build()),
+        Arguments.of(
+            "Can submerge but only defenseless transports on the defense",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .gameData(
+                    MockGameData.givenGameData()
+                        .withTransportCasualtiesRestricted(true)
+                        .withSubmersibleSubs(true)
+                        .build())
+                .defendingUnits(List.of(givenUnitTransport()))
+                .build()),
+        Arguments.of(
+            "No retreat territories and can not submerge",
+            givenBattleStateBuilder()
+                .gameData(
+                    MockGameData.givenGameData().withTransportCasualtiesRestricted(false).build())
+                .attackerRetreatTerritories(List.of())
+                .build()));
+  }
+
+  static class MockGameData {
+    private final GameData gameData;
+    private final GameProperties gameProperties;
+    private boolean propertiesSetup = false;
+
+    private MockGameData() {
+      gameData = mock(GameData.class);
+      gameProperties = mock(GameProperties.class);
+    }
+
+    static MockGameData givenGameData() {
+      return new MockGameData();
+    }
+
+    GameData build() {
+      return gameData;
+    }
+
+    MockGameData withTransportCasualtiesRestricted(final boolean value) {
```

Comment:
I like this method :+1: 

I tend to think this indicates we should aggregate more of this kind of logic in `GameData` itself so you could have method on game data to turn this property on.